### PR TITLE
Word count: create exercise

### PR DESCRIPTION
**Pull request by glennj**
This is the next core exercise following "series"

### CODE REVIEW

File: exercises/word-count/example.tcl [hunk 1]
```diff
@@ -0,0 +1,22 @@
+proc countWords {sentence} {
+
+    # A "word" is"
+    # 1. A _number_ composed of one or more ASCII digits (ie "0" or "1234") OR
+    # 2. A _simple word_ composed of one or more ASCII letters (ie "a" or "they") OR
+    # 3. A _contraction_ of two _simple words_ joined by a single apostrophe (ie "it's" or "they're")
```
Comment (sshine) [inline exercises/word-count/example.tcl]: We should not repeat the problem description in the example solution, just as we don't expect students to do so in their submissions. These comments are indeed helpful when put next to a regex, but perhaps we could shorten this to inlining comments in the regexp.

File: exercises/word-count/word-count.test [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+        word-count-13   "alternating word separators not detected as a word"
+        ",\n,one,\n ,two \n 'three'" 
+        {"one" 1 "two" 1 "three" 1}
+    }
+
+    # Implementation notes:
+    # I'm expecting the output to be a dictionary.
+    # A list with an even number of elements acts like a dictionary.
+    # `array get arrayVar` does as well.
```
Comment (sshine) [inline exercises/word-count/word-count.test]: We should either make it explicit in the stub that we expect a `dict`, or somehow make the test suite compatible with a generic interface such as a list of an even number of elements. Otherwise we force people to look in the test file to figure out how to implement the exercise. Having used the Perl track for training people at my day job, my experience with this is negative.

If we wish to make a comment about `dict` being preferred over `array`, or the two data structures being somewhat equivalent, or any other Tcl-specific notes we like to give, we should put those in `.meta/hints.md` so they're shown below the exercise description.

In light of this, the implementation notes above should also be removed.

File: exercises/word-count/example.tcl [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    set re {
+        \m (?:
+            \d+
+            |
+            [[:alpha:]]+ (?: ' [[:alpha:]]+ )?
+        ) \M
+    }
```
Comment (sshine) [inline exercises/word-count/example.tcl]: How about this instead?

```tcl
    set re {
        \m (?:
            \d+ |                               # digits
            [[:alpha:]]+ (?: ' [[:alpha:]]+ )?  # words, contractions
        ) \M
    }
```

File: exercises/word-count/word-count.tcl [hunk 4]
```diff
@@ -0,0 +1,3 @@
+proc countWords {sentence} {
+    throw NOT_IMPLEMENTED "Implement this procedure."
+}
```
Comment (sshine) [inline exercises/word-count/word-count.tcl]: How about the following?

```tcl
proc countWords {sentence} {
    set frequencies [dict create]
    throw NOT_IMPLEMENTED "Implement this procedure."
}
```

**Comment by glennj** [comment]
Great feedback, thanks.

I'd forgotten about the hints file. One day, when I'm in writing mode, I'll review the exercises for more opportunities for hints.
